lib/ui/zoomable_live_image.dart

```diff
@@ -95,37 +102,55 @@ class _ZoomableLiveImageState extends State<ZoomableLiveImage>
         aspectRatio: _videoPlayerController.value.aspectRatio,
         autoPlay: true,
         autoInitialize: true,
-        looping: false,
+        looping: true,
         allowFullScreen: false,
         showControls: false);
     return Chewie(controller: _chewieController);
   }
 
-  void _loadLiveVideo() {
-    // todo: add wrapper to download file from server if local is missing
-    getFile(widget.photo, liveVideo: true).then((file) {
-      if (file != null && file.existsSync()) {
-        _logger.fine("loading  from local");
-        _setVideoPlayerController(file: file);
-      } else if (widget.photo.uploadedFileID != null) {
-        _logger.fine("loading from remote");
-        getFileFromServer(widget.photo, liveVideo: true).then((file) {
-          if (file != null && file.existsSync()) {
-            _setVideoPlayerController(file: file);
-          } else {
-            _logger.warning("failed to load from remote" + widget.photo.tag());
-          }
-        });
-      }
+  Future<void> _loadLiveVideo() async {
+    // do nothing is already loading or loaded
+    if (_isLoadingVideoPlayer || _videoPlayerController != null) {
+      return;
+    }
+    _isLoadingVideoPlayer = true;
+    if (_file.isRemoteFile() && !(await isFileCached(_file, liveVideo: true))) {
+      showToast("downloading...", toastLength: Toast.LENGTH_LONG);
+    }
+
+    var videoFile = await getFile(widget.file, liveVideo: true)
+        .timeout(Duration(seconds: 15))
+        .onError((e, s) {
+      _logger.info("getFile failed ${_file.tag()}", e);
+      return null;
     });
+
+    if ((videoFile == null || !videoFile.existsSync()) &&
+        _file.isRemoteFile()) {
+      videoFile = await getFileFromServer(widget.file, liveVideo: true)
```

Comment:
`getFile` internally calls `getFileFromServer` too right? Why are we retrying here?

Reply:
This is to handle the case where we have a localID in the db but we fail to load the video from the local storage. (Saw similar pattern in ZoomableImageView widget).

Reply:
Got it. It's to handle files that were deleted on local.